Fix mapping in exports

uilton-oliveira opened the pull request:
Fixes #1019
- Add correct mapping for public entrypoints.

uilton-oliveira committed:
Fix mapping in exports

Fixes #1019
- Add correct mapping for public entrypoints.

claviska commented on package.json:
```diff
@@ -13,7 +13,13 @@
       "import": "./dist/shoelace.js"
-    }
+    },
+    "./shoelace.js": "./dist/shoelace.js",
```
Does this need to be here? Seems like a dup of line 15. Can either be removed?

uilton-oliveira replied:
let me test it

uilton-oliveira replied:
```bash
Module not found: Error: Package path ./shoelace.js is not exported from package C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system (see exports field in C:\Programacao\platform-cortex\modules\competitiva\frontend\node_modules\@cortex-intelligence\design-system\package.json)
```

Does not work without it... the import in the dot block probably is a typescript thing... not sure

claviska replied:
If I understand it correctly, the `.` export will pull from `dist/shoelace.js`:

```ts
import { thing } from '@shoelace-style/shoelace';
```

And the `./shoelace.js` export will also pull from the same place.

```ts
import { thing } from '@shoelace-style/shoelace/shoelace.js';
```

The latter seems redundant, but I'm not sure which is better. The only time one should pull from `shoelace.js` is when you want the entire library to load. Depending on the bundler, tree shaking may not work as expected (looking at you, webpack!)

So I wonder if we should discourage pulling from `@shoelace-style/shoelace` without specifying a specific file, be it `shoelace.js` (all components) or specific components/utils.

claviska replied:
> Does not work without it... the import in the dot block probably is a typescript thing...

Does it fail using:

```ts
import { thing } from '@shoelace-style/shoelace';
```

or this one?

```ts
import { thing } from '@shoelace-style/shoelace/shoelace.js';
```

uilton-oliveira replied:
Without the `"./shoelace.js": "./dist/shoelace.js",` entry:
`import { thing } from '@shoelace-style/shoelace';` works
`import { thing } from '@shoelace-style/shoelace/shoelace.js';` fail

Is it ok to remove it?

claviska replied:
Yeah that's fine. One way to do things is preferable, so that can be the main entrypoint and if treeshaking is an issue with a particular bundler/build tool, they can update it to use the full path.

uilton-oliveira replied:
Done

claviska replied:
Thanks, looks good. This _will_ mean that CDN paths will be out of sync since they still need the `dist` for jsDelivr, but once we move to our own CDN we can remap that too.

claviska replied:
Hmm, I'm actually not sure if CDN users will be able to remove `dist` now too. Might have to do a test release. The best hint I can find whether JSDelivr supports this or not is here:

https://github.com/jsdelivr/jsdelivr/issues/18298#issuecomment-989889555

Has anyone tried this before with jsDelivr?

uilton-oliveira replied:
Based on that comment, it should probably work... but never tested it... maybe open a Discussion to bring more visibility to this question?

claviska replied:
I got [some clarification](https://twitter.com/jsDelivr/status/1595110883482800129?s=20&t=ns59wOK2wWUF_0MWuCExKw):

> The exports field isn't considered in that case but you can of course load any file directly (via its full URL). However, we don't do any rewriting of imports in that case so it would only work if you don't have imports of other pages or use resolved URLs for them.

The previous comment was intended for [esm.run](https://www.jsdelivr.com/esm), which has a [known issue](https://github.com/jsdelivr/jsdelivr/issues/18337) with singletons so I can't use that at the moment.

After looking through the docs, I think leaving `dist` in both paths for now will be best to avoid confusion. When Shoelace moves to its own CDN, which is something I'd like to do, we can probably remap it there and remove `dist` from both at the same time.

I'll make this update shortly. Thanks again!

claviska replied:
Updated in 6c7d7f4b7ec344c90fb9b4050197d2aed3fa6332.

claviska reviewed (changes requested):
Looks good, but one question about a possible dup.

uilton-oliveira committed:
Remove shoelace.js from from exports list

Should still works fine while importing this way: import { thing } from '@shoelace-style/shoelace';
